### PR TITLE
feat: add go service build and test

### PR DESCRIPTION
**Pull request by dreinhardt-terminus**
- [x] I promise to make sure that this PR is correct before merging it. 

#### Does this change have potential to cause downtime? 

- [x] No - This is a low risk change
- [ ] Yes - This is high risk change and must be elevated or higher and you need more approvals

### CODE REVIEW

File: service-build/spark-test/action.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Install SSH Key
+      uses: webfactory/ssh-agent@v0.4.1
+      with:
+        ssh-private-key: ${{ inputs.ssh_key }}
+    - name: Pull
+      uses: actions/checkout@v2
+      with:
+        lfs: true
+    - name: package_spark_lib
+      uses: docker://getterminus/scala-sbt-ci-image
```
Comment (jcogilvie) [inline service-build/spark-test/action.yaml:32]: I'm vaguely skeptical about this.  why not use the GHA for SBT?  (surely there is one)

File: shell-scripts/input-validator.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+debug () {
```
Comment (jcogilvie) [inline shell-scripts/input-validator.sh:3]: Not necessarily needed at this time but I wonder if we ought to factor out these debug, error, etc functions into another shell script we can `source` into the others.

**Review by jcogilvie — APPROVED** [review]
Approved so long as we've done downstream testing of this and it works.

File: shell-scripts/input-validator.sh (hunk 2 above)
Comment (dreinhardt-terminus) [reply shell-scripts/input-validator.sh:3]: That's a good idea. I'll do that in a patch.

**Pull request merged**